src/lib/es.js

```diff
@@ -43,20 +43,28 @@ function buildDatetimeQuery(parameters) {
   let dateQuery
   const { datetime } = parameters
   if (datetime) {
-    const dataRange = datetime.split('/')
-    if (dataRange.length === 2) {
+    if (!datetime.includes('/')) {
```

Comment:
It's more verbose, but consider breaking this function apart into something like this. I think it makes it easier to understand what's going on.

```js
/**
 * @param {string} datetime
 * @returns {unknown}
 */
const buildDatetimeRange = (datetime) => {
  const [start, end, ...rest] = datetime.split('/')

  if (rest.length > 0) {
    return new Error('datetime value is invalid, contains more than one forward slash')
  }

  if (!start && !end) {
    return new Error(
      'datetime value is invalid, at least one end of the interval must be closed'
    )
  }

  if (start === '..' && end === '..') {
    return new Error(
      'datetime value is invalid, at least one end of the interval must be closed'
    )
  }

  let range

  if (start && end) range = { gte: start, lte: end }
  else if (start) range = { gte: start }
  else range = { lte: end }

  return {
    range: {
      'properties.datetime': range
    }
  }
}

/**
 * @param {string} datetime
 * @returns {unknown}
 */
const buildDatetimeTerm = (datetime) => ({
  term: {
    'properties.datetime': datetime
  }
})

/**
 * @param {string} datetime
 * @returns {boolean}
 */
const isExactDatetime = (datetime) => !datetime.includes('/')

/**
 * @param {Object} params
 * @param {string} params.datetime
 * @returns {unknown|Error|undefined}
 */
const buildDatetimeQuery = ({ datetime }) => {
  if (!datetime) return undefined

  return isExactDatetime(datetime)
    ? buildDatetimeTerm(datetime)
    : buildDatetimeRange(datetime)
}
```

Reply:
I chose not to do too much work on this right now because we need to significantly improve the datetime validation https://github.com/stac-utils/stac-server/issues/175